crates/pop-cli/src/commands/up/rollup.rs

```diff
@@ -237,6 +441,81 @@ fn prompt_for_proxy_address(cli: &mut impl Cli) -> Result<String> {
 	Ok(format!("Id({address})"))
 }
 
+// Prompts the user to select deployment options.
+fn prompt_provider(
+	skip_registration: bool,
+	cli: &mut impl Cli,
+) -> Result<Option<DeploymentProvider>> {
+	let mut predefined_action = cli.select("Select your deployment method:");
+	for action in DeploymentProvider::VARIANTS {
+		predefined_action = predefined_action.item(
+			Some(action.clone()),
+			action.name(),
+			format_url(action.base_url()),
+		);
+	}
+	if !skip_registration {
+		predefined_action = predefined_action.item(
+			None,
+			"Register",
+			"Register the rollup on the relay chain without deploying with a provider",
+		);
+	}
+	Ok(predefined_action.interact()?)
+}
+
+// Prompts user to select a supported chain for deployment.
+fn prompt_supported_chain(cli: &mut impl Cli) -> Result<&SupportedChains> {
+	let mut chain_selected = cli.select("Select a Relay Chain:");
+	for chain in SupportedChains::VARIANTS {
+		chain_selected = chain_selected.item(chain, chain.to_string(), "");
+	}
+	Ok(chain_selected.interact()?)
+}
+
+// Prompts for an API key and attempts to read from environment first.
+fn prompt_api_key(
+	env_var_name: &str,
+	provider: &DeploymentProvider,
+	cli: &mut impl Cli,
+) -> Result<String> {
+	if let Ok(api_key) = env::var(env_var_name) {
+		cli.info(format!("Using API key from environment variable ({env_var_name})."))?;
+		return Ok(api_key);
+	}
+	cli.warning(format!("No API key found for the environment variable `{env_var_name}`.\n{}\n{}", style(format!("You can generate an API key at: {}", style(provider.base_url().to_string()).underlined().magenta())), style(format!("Note: Consider setting this variable in your shell (e.g., `export {env_var_name}=...`) or system environment so you won’t be prompted each time.")).dim()))?;
+	let api_key = cli.password("Enter your API key:").interact()?;
+	Ok(api_key)
+}
+
+// Prompts the user to choose which template was used.
+fn prompt_template_used(cli: &mut impl Cli) -> Result<&str> {
+	cli.warning("We could not automatically detect which template was used to build your rollup.")?;
+	let mut template = cli.select("Select the template used:");
+	for supported_template in Parachain::VARIANTS.iter().filter_map(|variant| {
+		variant
+			.deployment_name()
+			.map(|name| (name, variant.name(), variant.description().trim()))
+	}) {
+		template = template.item(supported_template.0, supported_template.1, supported_template.2);
+	}
+	Ok(template.interact()?)
+}
+
+// Warns user about which templates are supported for the given provider.
+fn warn_supported_templates(provider: &DeploymentProvider, cli: &mut impl Cli) -> Result<()> {
+	let supported_templates: Vec<String> = Parachain::VARIANTS
+		.iter()
+		.filter_map(|variant| variant.deployment_name().map(|_| variant.name().to_string()))
+		.collect();
+	cli.warning(format!(
+		"Currently {} only supports the following templates: {}.\n",
+		provider.name(),
+		style(supported_templates.join(", ")).dim()
```

Comment:
Could we list the different chains in different lines instead of all of them printed in the same one ?

The list of templates feels already big enough to be just in one line.

Reply:
Improved in https://github.com/r0gue-io/pop-cli/pull/459/commits/3be0028549a651484df68081db03783da24066b9

<img width="614" alt="Captura de pantalla 2025-03-26 a las 9 26 49" src="https://github.com/user-attachments/assets/0e908616-0ae6-4abe-8f0f-5f44f0b154e9" />

Reply:
Way nicer to read, thanks :)